README.md

```diff
@@ -4,13 +4,32 @@ This repo contains the code for building http://source.dot.net
 ## Build Status
 [![Build Status](https://dev.azure.com/dnceng/internal/_apis/build/status/dotnet-source-indexer/dotnet-source-indexer%20CI?branchName=master)](https://dev.azure.com/dnceng/internal/_build/latest?definitionId=612&branchName=master)
 
-## Building
+## What Is It?
+This repo uses https://github.com/KirillOsenkov/SourceBrowser (with a few additions here https://github.com/dotnet/SourceBrowser/tree/source-indexer) to index the dotnet sources and produce a navigatable and searchable website containing the full source code. This includes code from the runtime, winforms, wpf, aspnetcore, and msbuild, among others. For a full list see here https://github.com/dotnet/source-indexer/blob/master/src/index/repositories.props.
 
-To clone and build locally (Windows only):
-1. `git clone --recursive https://github.com/dotnet/source-indexer.git`
-2. `cd source-indexer\src\SourceBrowser`
-3. `git fetch --all`
-4. `git checkout source-indexer`
-5. `dotnet restore`
-6. `cd ..\..`
-7. `msbuild build.proj`
+## Build Prerequsites
+The build requires .net core 3.1.201 and Visual Studio 2019 to build.
+
+## Build
+The build will only work on windows because the source indexer executable is a .net framework executable.
+1. `git clone https://github.com/dotnet/source-indexer.git`
+1. For each *.sln file `dotnet restore`
+1. Find VS 2019 msbuild.exe on your machine, typically found at `C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe`
+1. `msbuild build.proj`
+
+## Running the built index
+After the build is finished the index will exist in `bin\index` and can be run by running `dotnet Microsoft.SourceBrowser.SourceIndexServer.dll` in that folder. The index will be served on `http://localhost:5000`
+
+## Deployment
+The index is deployed by the VSTS build to the netsourceindex azure app service, with the index data stored in the netsourceindex storage account. The deployment does the following things.
```

Comment:
Where are the secrets kept?

Reply:
There are none other than a service connection to the azure subscription that is in VSTS.